### PR TITLE
[SYCL] Enable SPV_KHR_bit_instructions for SYCL code

### PR DESCRIPTION
**Pull request by stefanatwork**
SPV_KHR_bit_instructions is required when using __builtin_bitreverse32(). SYCL code that was compiled without that extension enabled and would fail when __builtin_bitreverse32() were present in source.

### CODE REVIEW

**Comment by stefanatwork** [comment]
Example of code that would not compile:
```
#include <sycl/sycl.hpp>

using namespace sycl;

int main(int argc, char **argv)
{
    queue q;
    int *out = malloc_shared<int>(1024, q);
    q.submit([&](handler &h) {
        h.parallel_for(range<1>(1024), [=](sycl::id<1> i){
            out[i] = __builtin_bitreverse32(i);
        });
    });
    free(out);
    return 0;
}
```

**Comment by stefanatwork** [comment]
This used to work commits 85acfa24169fe47fc8b23ada866c42ba0a587647

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -10188,7 +10188,7 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
     std::string DefaultExtArg =
         ",+SPV_EXT_shader_atomic_float_add,+SPV_EXT_shader_atomic_float_min_max"
         ",+SPV_KHR_no_integer_wrap_decoration,+SPV_KHR_float_controls"
-        ",+SPV_KHR_expect_assume,+SPV_KHR_linkonce_odr";
+        ",+SPV_KHR_expect_assume,+SPV_KHR_linkonce_odr,+SPV_KHR_bit_instructions";
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp:10191]: Can you also update `clang/test/Driver/sycl-spirv-ext.c`?

Comment (ph0b) [reply clang/lib/Driver/ToolChains/Clang.cpp:10191]: @mdtoguchi Stefan is out at the moment and I don't have write permissions, could you make that update ?

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp:10191]: Hey it worked!  done :)

**Review by mdtoguchi — APPROVED** [review]

File: clang/lib/Driver/ToolChains/Clang.cpp (hunk 1 above)
Comment (ph0b) [reply clang/lib/Driver/ToolChains/Clang.cpp:10191]: @mdtoguchi perfect! if it's approved and ready-to-be-merged, can you merge it ? I don't have write permissions for this either.

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this should be ready for merge - thanks!

**Pull request merged**